### DISCUSSION TITLE
Separate enum suffixes, How to achieve 1.a) enumerations?

### DISCUSSION THREAD

**Discussion by frank-ruegamer** [typst/typst#1235] [Q&A] [answered]
I would like to define numbered lists like the following
```
1. First
  a) This
  b) That
2. Second
```

Unfortunately the pattern `1.a)` will add the suffix `)` to each level. 
Using a function I came as far as
```typst
#set enum(numbering: (..n) => {
	if n.pos().len() > 1 {
		str(n.pos().last()) + ")"
	} else {
		str(n.pos().last()) + "."
	}
	}, full: true)
```

but that will create 
```
1. First
  1) This
```
The missing step would be converting `1` to `a` and so on.

**Comment by PgBiel** [comment, marked as answer]
> The missing step would be converting 1 to a and so on.

You can use `numbering("a.", 1)` (for example) to achieve the same output that you'd have with the given numbering string, but on code. See the docs here: https://typst.app/docs/reference/meta/numbering/

For a working example of what you need, see https://github.com/typst/typst/issues/905#issuecomment-1528829219

**Reply by frank-ruegamer** [reply]
Thank you very much! I ended up on this, which I am happy with
```typst
#set enum(full: true, numbering: (..n) => {
	let format = if n.pos().len() > 1 {"a)"} else {"1."}
	numbering(format, n.pos().last())
})
```
